test/test_io.py

```diff
@@ -64,234 +64,290 @@ def temp_video(num_frames, height, width, fps, lossless=False, video_codec=None,
-            video_info = io._probe_video_from_memory(filebuffer)
-            self.assertAlmostEqual(video_info.video_duration, 2, delta=0.1)
-            self.assertAlmostEqual(video_info.video_fps, 5, delta=0.1)
-
-    def test_read_timestamps(self):
-        with temp_video(10, 300, 300, 5) as (f_name, data):
-            pts, _ = io.read_video_timestamps(f_name)
-            # note: not all formats/codecs provide accurate information for computing the
-            # timestamps. For the format that we use here, this information is available,
-            # so we use it as a baseline
-            container = av.open(f_name)
-            stream = container.streams[0]
-            pts_step = int(round(float(1 / (stream.average_rate * stream.time_base))))
-            num_frames = int(round(float(stream.average_rate * stream.time_base * stream.duration)))
-            expected_pts = [i * pts_step for i in range(num_frames)]
-
-            self.assertEqual(pts, expected_pts)
-            container.close()
-
-    def test_read_partial_video(self):
-        with temp_video(10, 300, 300, 5, lossless=True) as (f_name, data):
-            pts, _ = io.read_video_timestamps(f_name)
-            for start in range(5):
-                for offset in range(1, 4):
-                    lv, _, _ = io.read_video(f_name, pts[start], pts[start + offset - 1])
-                    s_data = data[start:(start + offset)]
-                    self.assertEqual(len(lv), offset)
-                    assert_equal(s_data, lv)
-
-            if get_video_backend() == "pyav":
-                # for "video_reader" backend, we don't decode the closest early frame
-                # when the given start pts is not matching any frame pts
-                lv, _, _ = io.read_video(f_name, pts[4] + 1, pts[7])
-                self.assertEqual(len(lv), 4)
-                assert_equal(data[4:8], lv)
-
-    def test_read_partial_video_bframes(self):
-        # do not use lossless encoding, to test the presence of B-frames
-        options = {'bframes': '16', 'keyint': '10', 'min-keyint': '4'}
-        with temp_video(100, 300, 300, 5, options=options) as (f_name, data):
-            pts, _ = io.read_video_timestamps(f_name)
-            for start in range(0, 80, 20):
-                for offset in range(1, 4):
-                    lv, _, _ = io.read_video(f_name, pts[start], pts[start + offset - 1])
-                    s_data = data[start:(start + offset)]
-                    self.assertEqual(len(lv), offset)
-                    assert_equal(s_data, lv, rtol=0.0, atol=self.TOLERANCE)
+# compression adds artifacts, thus we add a tolerance of
+# 6 in 0-255 range
+TOLERANCE = 6
+
+
+@pytest.mark.skipif(get_video_backend() != "pyav" and not io._HAS_VIDEO_OPT,
+                    reason="video_reader backend not available")
+@pytest.mark.skipif(av is None, reason="PyAV unavailable")
+def test_write_read_video():
+    with temp_video(10, 300, 300, 5, lossless=True) as (f_name, data):
```

Comment:
In the future, we could change `temp_video` into a fixture. But let's keep this for another PR

Reply:
@NicolasHug can you open an issue to track this? Making this into a fixture will make tests faster

Reply:
I'll keep this in the back of my mind but I'd like to focus on porting the rest of the tests for now. The fixture will need to be parametrized somehow and while I think it's possible, I might be missing something and I'm not 100% sure.

There won't be any significant time gain, `temp_video` is extremely cheap. I created this test:
```py
def test_lol():
    for _ in range(1000):
        options = {'bframes': '16', 'keyint': '10', 'min-keyint': '4'}
        temp_video(10, 300, 300, 5, lossless=True, options=options)
```
(with and without options, it was the same)

and it ran in less than 5ms. For ref, the slowest tests are:

```
3.52s call     test/test_io.py::TestVideo::test_write_video_with_audio
0.43s call     test/test_io.py::TestVideo::test_read_partial_video_bframes[1-60]
0.41s call     test/test_io.py::TestVideo::test_read_partial_video_bframes[2-60]
0.41s call     test/test_io.py::TestVideo::test_read_partial_video_bframes[2-20]
0.40s call     test/test_io.py::TestVideo::test_read_partial_video_bframes[3-0]
0.40s call     test/test_io.py::TestVideo::test_read_partial_video_bframes[2-40]
0.40s call     test/test_io.py::TestVideo::test_read_partial_video_bframes[3-20]
0.39s call     test/test_io.py::TestVideo::test_read_partial_video_bframes[3-60]
0.39s call     test/test_io.py::TestVideo::test_read_partial_video_bframes[2-0]
0.39s call     test/test_io.py::TestVideo::test_read_partial_video_bframes[3-40]
0.37s call     test/test_io.py::TestVideo::test_read_partial_video_bframes[1-40]
0.37s call     test/test_io.py::TestVideo::test_read_partial_video_bframes[1-0]
0.36s call     test/test_io.py::TestVideo::test_read_partial_video_bframes[1-20]
0.11s call     test/test_io.py::TestVideo::test_read_partial_video_pts_unit_sec[1-1]
0.10s call     test/test_io.py::TestVideo::test_read_partial_video_pts_unit_sec[2-1]
0.10s call     test/test_io.py::TestVideo::test_read_partial_video_pts_unit_sec[3-1]
0.10s call     test/test_io.py::TestVideo::test_read_partial_video[2-3]
0.10s call     test/test_io.py::TestVideo::test_read_partial_video[1-3]
0.09s call     test/test_io.py::TestVideo::test_read_partial_video[3-3]
0.09s call     test/test_io.py::TestVideo::test_read_partial_video[1-2]
```

Reply:
Ok sounds good. If the video creation is cheap we don't need to bother about it, thanks for double-checking!